### DISCUSSION TITLE
CouchDB upgrade problems from old BB version

### DISCUSSION THREAD

**Discussion by maiducan** [Budibase/budibase#10600] [Help]
Hello,

my couchdb image is unable to load correctly on selfhosted instance. Down is the docker log of couchdb container. Due to this fact, bbapps cannot connect and resulting in Not Found blank page. Can anybody help?

I have tried all different image versions of couchdb and other bb images,  ... only clean install will solve this problem, but i need to use my current volumes ..

`Crash dump is being written to: erl_crash.dump...[info] 2023-05-16T05:13:18.610685Z couchdb@127.0.0.1 <0.255.0> -------- Preflight check: Asserting Admin Account

[info] 2023-05-16T05:13:18.613005Z couchdb@127.0.0.1 <0.255.0> -------- Apache CouchDB 3.2.1 is starting.

[info] 2023-05-16T05:13:18.613065Z couchdb@127.0.0.1 <0.256.0> -------- Starting couch_sup
[info] 2023-05-16T05:13:18.706162Z couchdb@127.0.0.1 <0.255.0> -------- Apache CouchDB has started. Time to relax.

[notice] 2023-05-16T05:13:18.711119Z couchdb@127.0.0.1 <0.303.0> -------- rexi_server : started servers
[notice] 2023-05-16T05:13:18.711977Z couchdb@127.0.0.1 <0.307.0> -------- rexi_buffer : started servers
[error] 2023-05-16T05:13:18.726754Z couchdb@127.0.0.1 <0.317.0> -------- Could not open file ./data/_nodes.couch: permission denied
[info] 2023-05-16T05:13:18.726845Z couchdb@127.0.0.1 <0.316.0> -------- open_result error {error,eacces} for _nodes
[error] 2023-05-16T05:13:18.727211Z couchdb@127.0.0.1 <0.313.0> -------- Supervisor mem3_sup had child mem3_nodes started with mem3_nodes:start_link() at undefined exit with reason no case clause matching {error,eacces} at mem3_util:ensure_exists/1(line:298) <= mem3_nodes:initialize_nodelist/0(line:94) <= mem3_nodes:init/1(line:45) <= gen_server:init_it/2(line:417) <= gen_server:init_it/6(line:385) <= proc_lib:init_p_do_apply/3(line:226) in context start_error
[error] 2023-05-16T05:13:18.727500Z couchdb@127.0.0.1 <0.313.0> -------- Supervisor mem3_sup had child mem3_nodes started with mem3_nodes:start_link() at undefined exit with reason no case clause matching {error,eacces} at mem3_util:ensure_exists/1(line:298) <= mem3_nodes:initialize_nodelist/0(line:94) <= mem3_nodes:init/1(line:45) <= gen_server:init_it/2(line:417) <= gen_server:init_it/6(line:385) <= proc_lib:init_p_do_apply/3(line:226) in context start_error
[error] 2023-05-16T05:13:18.727868Z couchdb@127.0.0.1 <0.311.0> -------- CRASH REPORT Process  (<0.311.0>) with 0 neighbors exited with reason: {{shutdown,{failed_to_start_child,mem3_nodes,{{case_clause,{error,eacces}},[{mem3_util,ensure_exists,1,[{file,"src/mem3_util.erl"},{line,298}]},{mem3_nodes,initialize_nodelist,0,[{file,"src/mem3_nodes.erl"},{line,94}]},{mem3_nodes,init,1,[{file,"src/mem3_nodes.erl"},{line,45}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,417}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,385}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}},{mem3_app,start,[normal,[]]}} at application_master:init/4(line:138) <= proc_lib:init_p_do_apply/3(line:226); initial_call: {application_master,init,['Argument__1','Argument__2',...]}, ancestors: [<0.310.0>], message_queue_len: 1, links: [<0.310.0>,<0.16.0>], dictionary: [], trap_exit: true, status: running, heap_size: 987, stack_size: 28, reductions: 227
[error] 2023-05-16T05:13:18.728077Z couchdb@127.0.0.1 <0.311.0> -------- CRASH REPORT Process  (<0.311.0>) with 0 neighbors exited with reason: {{shutdown,{failed_to_start_child,mem3_nodes,{{case_clause,{error,eacces}},[{mem3_util,ensure_exists,1,[{file,"src/mem3_util.erl"},{line,298}]},{mem3_nodes,initialize_nodelist,0,[{file,"src/mem3_nodes.erl"},{line,94}]},{mem3_nodes,init,1,[{file,"src/mem3_nodes.erl"},{line,45}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,417}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,385}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}},{mem3_app,start,[normal,[]]}} at application_master:init/4(line:138) <= proc_lib:init_p_do_apply/3(line:226); initial_call: {application_master,init,['Argument__1','Argument__2',...]}, ancestors: [<0.310.0>], message_queue_len: 1, links: [<0.310.0>,<0.16.0>], dictionary: [], trap_exit: true, status: running, heap_size: 987, stack_size: 28, reductions: 227
[info] 2023-05-16T05:13:18.728242Z couchdb@127.0.0.1 <0.16.0> -------- Application mem3 exited with reason: {{shutdown,{failed_to_start_child,mem3_nodes,{{case_clause,{error,eacces}},[{mem3_util,ensure_exists,1,[{file,"src/mem3_util.erl"},{line,298}]},{mem3_nodes,initialize_nodelist,0,[{file,"src/mem3_nodes.erl"},{line,94}]},{mem3_nodes,init,1,[{file,"src/mem3_nodes.erl"},{line,45}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,417}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,385}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}},{mem3_app,start,[normal,[]]}}
[info] 2023-05-16T05:13:18.728422Z couchdb@127.0.0.1 <0.16.0> -------- Application mem3 exited with reason: {{shutdown,{failed_to_start_child,mem3_nodes,{{case_clause,{error,eacces}},[{mem3_util,ensure_exists,1,[{file,"src/mem3_util.erl"},{line,298}]},{mem3_nodes,initialize_nodelist,0,[{file,"src/mem3_nodes.erl"},{line,94}]},{mem3_nodes,init,1,[{file,"src/mem3_nodes.erl"},{line,45}]},{gen_server,init_it,2,[{file,"gen_server.erl"},{line,417}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,385}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,226}]}]}}},{mem3_app,start,[normal,[]]}}`

**Comment by shogunpurple** [comment]
Moved this to a discussion. @maiducan you need to give us more information. We don't even know what version of budibase you are on.

**Reply by maiducan** [reply]
I am on 2.2.3 cross all services.
Howerver i have tried the latest or other in between, and it always bubbles into this issue ..

 .. This setting is working on prod server. Last time i updated to the latest version of bb and migrated to backup server (18 days ago), it worked.  

But currently it doesn't - even if i update everything to the newest and migrate (same steps as I do every time).

Maybe there was some older budibase/couchdb image that worked, and now with some bb/couchdb update, it suddenly crashes.

On the new machine, I am doing docker-compose down (clears everything) and launching it back, but still not working..

I need to keep all the old volumes and have the exact states as on prod server .. So clean install doesn't do it for me.
